modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/java/enricher/DynamicMetadataDeclarationEnricher.java

```diff
@@ -281,6 +289,10 @@ public org.mule.sdk.api.metadata.resolving.AttributesTypeResolver getAttributesR
           return new org.mule.sdk.api.metadata.NullMetadataResolver();
         }
       };
+      addQueryModelProperties(declaration, query);
+      declareDynamicType(declaration.getOutput());
+      declareMetadataKeyId(declaration, null);
```

Comment:
Why did these lines need to change the placement?

Reply:
To use the metadataScope defined before and avoid using instanceOf on parseMetadataKeyAnnotations